### ISSUE TITLE
[com.sprint.trs and others] Unable to extract the trust manager / GmsProviderInstaller Did not insert the new SSL provider

### ISSUE THREAD

**Issue by jesse-git** [microg/GmsCore#1247]
**Affected app**
Name: Sprint IP Relay
Package id: com.sprint.trs
Version: 60.30.52

**Describe the bug**
```java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.platform.AndroidPlatform@6b26462, sslSocketFactory is class org.conscrypt.OpenSSLSocketFactoryImpl```

**To Reproduce**
Try to launch and login
1. Press Allow for push notifications
1. Attempt to log in
(Only deaf or hard of hearing or those with disabilities are allowed to legally make accounts. We are willing to test if necessary.)

**Expected behavior**
Start and log in

**System**
Android Version: 9

**microG**
microG Core version: 0.2.13.203915
microG Self-Check results: All checkboxes

**Comment by jesse-git** [comment]
Newer versions crash on startup with a more detailed error:

```
java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.platform.AndroidPlatform@6b26462, sslSocketFactory is class org.conscrypt.OpenSSLSocketFactoryImpl
stacktrace: java.lang.RuntimeException: Unable to create application com.sprint.trs.SprintIPRelayApplication: java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.platform.AndroidPlatform@6b26462, sslSocketFactory is class org.conscrypt.OpenSSLSocketFactoryImpl
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5917)
	at android.app.ActivityThread.access$1100(ActivityThread.java:197)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1653)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6672)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
Caused by: java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.platform.AndroidPlatform@6b26462, sslSocketFactory is class org.conscrypt.OpenSSLSocketFactoryImpl
	at okhttp3.internal.platform.Platform.buildCertificateChainCleaner(Unknown Source:41)
	at okhttp3.OkHttpClient$Builder.sslSocketFactory(Unknown Source:16)
	at com.sprint.trs.b.f.a(Unknown Source:134)
	at com.sprint.trs.b.c.a(Unknown Source:2)
	at com.sprint.trs.b.g.c.<init>(Unknown Source:10)
	at com.sprint.trs.SprintIPRelayApplication.g(Unknown Source:6)
	at com.sprint.trs.core.userinfodata.interactor.UserInfoInteractor.<init>(Unknown Source:3)
	at com.sprint.trs.a.a.<init>(Unknown Source:17)
	at com.sprint.trs.SprintIPRelayApplication.onCreate(Unknown Source:73)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5912)
	... 8 more
```

**Comment by jesse-git** [comment]
TLDR: Skip to bottom

Others have had the same error "unable to extract the trust manager":
* Issue #1043 "solutions":
    * "resolved" by uninstalling micro G. But sprint app failed with a different error. It really needs the GCM working. It relies on GCM to receive incoming calls.
    * We will not be trying the patched Russian version recommended because of security.
* Issue [#1045](https://github.com/microg/android_packages_apps_GmsCore/issues/1045#issuecomment-614711408), solutions:
    * Same as above: It really needs the GCM working.
    * I use the microg module in magisk, with the same magisk version mentioned in the other issue. One solution was to enable magiskhide. My magiskhide was always disabled above. However, When enabled with most recent sprint app version, it makes no difference.
* Issue #1005 solutions:
    * Use an older (Sprint/whatever) app version, which I mentioned when opening the issue this does not work because it produces a slightly different error.
    * Disabling micro g, except reenabling immediately after app starts, in the case of the sprint app, before logging in. THIS WORKS. Can we get this as a permanent solution? It is a pain 2 do.

**Comment by jesse-git** [comment]
Maybe this and other packages need to be disabled like #1241 ?

**Comment by jesse-git** [comment]
It is fixed on 0.2.14
